projects/core/src/routing/event/routing-event.builder.ts

```diff
@@ -0,0 +1,164 @@
+import { Injectable } from '@angular/core';
+import { ofType } from '@ngrx/effects';
+import { RouterNavigatedAction, ROUTER_NAVIGATED } from '@ngrx/router-store';
+import { ActionsSubject } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { filter, map, switchMap, withLatestFrom } from 'rxjs/operators';
+import { EventService } from '../../event/event.service';
+import { ProductSearchService } from '../../product/facade/product-search.service';
+import { ProductService } from '../../product/facade/product.service';
+import { createFrom } from '../../util/create-from';
+import { SemanticPathService } from '../configurable-routes';
+import { PageContext } from '../models/page-context.model';
+import { ActivatedRouterStateSnapshot } from '../store/routing-state';
+import {
+  CartPageVisited,
+  CategoryPageVisited,
+  HomePageVisited,
+  SearchPageVisited,
+  OrderConfirmationPageVisited,
+  PageVisited,
+  ProductDetailsPageVisited,
+} from './routing.events';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RoutingEventBuilder {
```

Comment:
Major: The Routing Module is meant to provide only generic routing mechanism. Ideally should not know about any specific pages (home, cart, search, ...). Especially it should not depend on (import) facades of specific feature modules (like ProductService or ProductSearchService). IMO feature-page specific *events* should be defined in feature-page specific modules.

It's debatable whether those feature-page modules should be part of the core or the storefront lib. We already have it inconsistent:
- we already have ProductPageModule and ProductListingPageModule in the storefront lib
- we already have ProductPageMetaResolver and CategoryPageMetaResolver in the core lib

Personally I'm in favour of moving all page-specific code to the storefront lib, but I'm happy to discuss it.

Nevertheless, in RoutingModule we should expose only generic router events, like `RouterNavigatedEvent` event. And then create other feature-page events on top of this event (using `eventService.get(RouterNavigatedEvent)` as a source observable).

Reply:
It appears this has been moved to another PR